PocketKit/Sources/PocketKit/Premium/PremiumUpgradeView.swift

```diff
@@ -0,0 +1,228 @@
+        HStack(spacing: 16) {
+            Image(asset: .checkMini).renderingMode(.template).foregroundColor(Color(.ui.teal3))
+            Text(text).style(.featureRow)
+            Spacer()
+        }.listRowSeparator(.hidden)
+    }
+}
+
+private struct PremiumUpgradeButton: View {
+    private let text: String
+    private let pricing: String
+    private let isYearly: Bool
+
+    init(text: String, pricing: String, isYearly: Bool) {
+        self.text = text
+        self.pricing = pricing
+        self.isYearly = isYearly
+    }
+
+    var body: some View {
+        if isYearly {
+            Button(action: {}) {
+                VStack(spacing: 8) {
+                    Text(text)
+                        .style(.yearlyPremiumRow)
+                    Text(pricing)
+                        .style(.yearlyPricing)
+                }
+                .padding()
+                .frame(maxWidth: .infinity, minHeight: 53)
+                .background(
+                    RoundedRectangle(cornerRadius: 4, style: .continuous)
+                        .fill(Color(.ui.coral2))
+                )
+            }
+        } else {
+            Button(action: {}) {
+                VStack(spacing: 8) {
+                    Text(text)
+                        .style(.monthlyPremiumRow)
+                    Text(pricing)
+                        .style(.pricing)
+                }
+                .padding()
+                .frame(maxWidth: .infinity, minHeight: 53)
+                .overlay(
+                    RoundedRectangle(cornerRadius: 4)
+                        .stroke(Color(.ui.grey1), lineWidth: 2)
+                )
+            }
+        }
+    }
+}
+
+private struct PremiumYearlyPercent: View {
+    var body: some View {
+        VStack {
+            Text("Save 25%")
+                .style(.percentSaved)
+                .multilineTextAlignment(.center)
+        }
+        .frame(width: 60, height: 60, alignment: .center)
+        .background(Circle().fill(Color(.ui.teal3)))
+    }
+}
+
+private struct PremiumInfoView: View {
+    private let text = """
+Subscriptions will be charged to your credit card through your iTunes account. \
+Your account will be charged $TBD (monthly) or $TBD (yearly) for renewal within 24 hours prior to the end of the current period. \
+Subscriptions will automatically renew unless canceled at least 24 hours before the end of the current period. \
+It will not be possible to immediately cancel a subscription. \
+You can manage subscriptions and turn off auto-renewal by going to your account settings after purchase. \
+Refunds are not available for unused portions of a subscription.
+"""
+
+    var body: some View {
+        Text(text).style(.info)
+    }
+}
+
+private struct PremiumTermsView: View {
+    var body: some View {
+        HStack(spacing: 16) {
+            Button { } label: { Text("Privacy Policy").style(.terms) }
+            Button { } label: { Text("Terms of Service").style(.terms) }
```

Comment:
what do we think of having all the hard coded strings in one file and pass it in so that we have one area to modify copy + will align with our localization effort. Should we also add `TODO: Localization`?

Reply:
yup, true. I just dont know where the localization work is at in its process.

Reply:
there is a PR, going to take a look at that later, but maybe consolidate and add `TODO:` similar to our empty states for now? not blocker - just a suggestion

Reply:
I am also debating on whether to keep my hardcoded strings in the `SearchView` similar to you, so if you prefer to keep it as is for now and defer to localization, I'm okay with that.

Reply:
your approval on my PR seems like we're okay with as-is for now.